File: identhub/src/main/java/de/solarisbank/identhub/session/data/verification/phone/VerificationPhoneRetrofitDataSource.java
package de.solarisbank.identhub.session.data.verification.phone;

import de.solarisbank.identhub.session.data.TransactionAuthenticationNumber;
import de.solarisbank.identhub.session.data.verification.phone.model.VerificationPhoneResponse;
import io.reactivex.Single;

public class VerificationPhoneRetrofitDataSource implements VerificationPhoneNetworkDataSource {

    private final VerificationPhoneApi verificationPhoneApi;

    public VerificationPhoneRetrofitDataSource(final VerificationPhoneApi verificationPhoneApi) {
        this.verificationPhoneApi = verificationPhoneApi;
    }

    @Override
    public Single<VerificationPhoneResponse> postAuthorize() {
        return verificationPhoneApi.postAuthorize();
    }

    @Override
    public Single<VerificationPhoneResponse> postConfirm(final TransactionAuthenticationNumber tan) {
        return verificationPhoneApi.postConfirm(tan);
    }
}

File: phone/src/main/java/de/solarisbank/identhub/phone/data/VerificationPhoneRetrofitDataSource.kt
package de.solarisbank.identhub.phone.data

import de.solarisbank.identhub.phone.model.TransactionAuthenticationNumber
import de.solarisbank.identhub.phone.model.VerificationPhoneResponse
import io.reactivex.Single

class VerificationPhoneRetrofitDataSource(private val verificationPhoneApi: VerificationPhoneApi) :
    VerificationPhoneNetworkDataSource {
    override fun postAuthorize(): Single<VerificationPhoneResponse> {
        return verificationPhoneApi.postAuthorize()
    }

    override fun postConfirm(tan: TransactionAuthenticationNumber): Single<VerificationPhoneResponse> {
        return verificationPhoneApi.postConfirm(tan)
    }
}